sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RedirectPolicy.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A HttpPipeline policy that redirects when an HTTP Redirect is received as response.
+ */
+public final class RedirectPolicy implements HttpPipelinePolicy {
+
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+
+    // Based on Stamp specific redirects design doc
+    private static final int MAX_REDIRECT_ATTEMPTS = 10;
```

Comment:
Let's remove max attempts from the policy as it should be handled by the strategy. This will need to have an update where the default strategy has an empty constructor.